util/rewrite_all_hw.py

```diff
@@ -0,0 +1,310 @@
+
+def step2(project_root, commit):
+    for (_ippath, info) in all_ips.items():
+        ippath = project_root / Path(_ippath)
+        ip_name = ippath.name
+        is_template = info.get("is_template", False)
+
+        replacements = {}
+        for typ in ["c", "rust"]:
+            new_target = f"//hw/top:{ip_name}_{typ}_regs"  # noqa: E231
+            if is_template:
+                for top in info["tops"]:
+                    # Some exceptions
+                    if ip_name in ["rv_plic", "alert_handler"]:
+                        replacements[f"//hw/top_{top}:{ip_name}_{typ}_regs"] = new_target  # noqa: E231, E501
+                    else:
+                        replacements[f"//hw/top_{top}/ip_autogen/{ip_name}:{ip_name}_{typ}_regs"] = new_target  # noqa: E231, E501
+            else:
+                replacements[f"//{_ippath}/data:{ip_name}_{typ}_regs"] = new_target  # noqa: E231
+
+        for (old, new) in replacements.items():
+            global_replace(project_root, old, new, verbose=False)
+
+    # Run buildifier.
+    run_buildifier(project_root)
+    run_topgen(project_root)
+
+    if commit:
+        subprocess.run(
+            [
+                "git", "commit", "-vas", "-m",
+                "Replace old header targets by new ones",  # noqa: E231
+                "-m", "This commit was generated by ./util/rewrite_all_hw.py --step=2",
+            ],
+            check=True,
+            cwd = project_root,
+        )
+
+
+def step3(project_root, commit):
+    files = list(set(
+        find_all_files(project_root, "autogen_hjson_c_header\\(") +
+        find_all_files(project_root, "autogen_hjson_rust_header\\(")
+    ))
+    print(files)
+    for path in files:
+        build_file_path = project_root / path
+        build_file = build_file_path.read_text().splitlines(keepends=True)
+        # Remove load to //rules:autogen.bzl
+        build_file, _, _ = delete_rule(
+            build_file,
+            'load',
+            '"//rules:autogen.bzl"',
+            build_file_path
+        )
+        # Remove autogen_hjson_c_header and autogen_hjson_rust_header
+        build_file = delete_all_rules(
+            build_file,
+            'autogen_hjson_c_header',
+            None,
+            build_file_path
+        )
+        build_file = delete_all_rules(
+            build_file,
+            'autogen_hjson_rust_header',
+            None,
+            build_file_path
+        )
+        build_file_path.write_text(''.join(build_file))
+
+    # Run buildifier.
+    run_buildifier(project_root)
+    run_topgen(project_root)
+
+    if commit:
+        subprocess.run(
+            [
+                "git", "commit", "-vas", "-m",
+                "Remove old header targets",  # noqa: E231
+                "-m", "This commit was generated by ./util/rewrite_all_hw.py --step=3",
+            ],
+            check=True,
+            cwd = project_root,
+        )
+
+
+def main():
+    parser = argparse.ArgumentParser()
```

Comment:
Adding args like this to ArgumentParser should print the doc string in the help message:
description=__doc__, formatter_class=RawTextHelpFormatter